### DISCUSSION TITLE
tutorials/intro-dgs/07-listings-rest-api

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#156] [course feedback]
# tutorials/intro-dgs/07-listings-rest-api

https://www.apollographql.com/tutorials/intro-dgs/07-listings-rest-api

**Comment by boolivar** [comment]
Consider using `RestClient.Builder` provided by spring boot - it is preconfigured with default `ObjectMapper` (in `HttpMessageConverters`) which has `DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES` feature disabled by default.
That way data retrieval becomes straightforward and natural for `RestClient` without handling unknown fields, nulls and `IOException`:
```java
@Component
public class ListingService {

    private static final String LISTING_API_URL = "https://rt-airlock-services-listing.herokuapp.com";

    private final RestClient client;

    public ListingService(RestClient.Builder builder) {
        client = builder.baseUrl(LISTING_API_URL).build();
    }

    public List<ListingModel> featuredListingsRequest() {
        return client
            .get()
            .uri("/featured-listings")
            .retrieve()
            .body(new ParameterizedTypeReference<List<ListingModel>>() {});
    }
}
```
Thank you for great tutorials! :rocket:
